osquery/events/linux/auditdnetlink.cpp

```diff
@@ -417,11 +417,14 @@ bool AuditdNetlinkReader::configureAuditService() noexcept {
     LOG(ERROR) << "Failed to install the audit rule due to one or more "
                << "syscalls with error " << audit_errno_to_name(rule_add_error)
                << ", Audit-based tables may not function as expected";
+
   } else if (FLAGS_audit_debug) {
     VLOG(1) << "Audit rule installed for all queued syscalls";
   }
 
-  if (FLAGS_audit_force_unconfigure) {
+  if (FLAGS_audit_force_reconfigure || rule_add_error >= 0) {
```

Comment:
`FLAGS_audit_force_reconfigure` should be `FLAGS_audit_force_unconfigure`.
Reconfigure is a "stronger" operation than unconfigure and it lists all the rules currently active on it's own and removes them.
Unconfigure is meant to remove all rules that osquery has ever added, even if they have not been added in the current run. But rules that have been added by an administrator should not be touched.

Reply:
Aah, the autocomplete, I had intended to write `unconfigure`, I would have accepted this suggestion from the autocomplete. I should have been more careful. Thank you for the review. I have updated the pull request.